### PR TITLE
Shield Repair

### PR DESCRIPTION
**Pull request by InterroLouis**
## About The Pull Request

Enables the repairing of shields, namely riot shields. They have integrity already, but were unable to be repaired. 

## Why It's Good For The Game

Now you can fix shields. 

## Changelog
:cl:
add: Riot Shields Now Repairable
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -2. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

File: code/game/objects/items/weapons/shields.dm [hunk 1]
```diff
@@ -43,7 +43,80 @@
...
+
+	var/obj/item/tool/weldingtool/WT = I
+
+	if(!WT.isOn())
+		return FALSE
+
+	for(var/obj/effect/xenomorph/acid/A in loc)
+		if(A.acid_t == src)
+			to_chat(user, "You can't get near that, it's melting!")
+			return TRUE
```
Comment (psykzz) [inline code/game/objects/items/weapons/shields.dm:83]: Every object has `current_acid` which will be set when acid is active on it.
Better than having to loop through all the items in a turf.

I also added a warning span around it to give it red text.
```suggestion
	if(current_acid)
		to_chat(user, "<span class='warning'>You can't get near that, it's melting!<span>")
		return TRUE
```

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-1). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

File: code/game/objects/items/weapons/shields.dm [hunk 2]
```diff
@@ -43,7 +43,79 @@
...
+		if(obj_integrity > max_integrity * 0.2)
+			return
+
+		if(metal_sheets.get_amount() < 1)
+			to_chat(user, "<span class='warning'>You need one metal sheet to repair the base of [src].</span>")
+			return
+
+		visible_message("<span class='notice'>[user] begins to repair the base of [src].</span>")
+
+		if(!do_after(user, 2 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY) || obj_integrity >= max_integrity)
```
Comment (Rohesie) [inline code/game/objects/items/weapons/shields.dm:61]: There should be an `action_busy` check before every `do_after`

File: code/game/objects/items/weapons/shields.dm [hunk 3]
```diff
@@ -43,7 +43,79 @@
 			to_chat(user, "<span class='notice'>It appears in decent condition, with some damage marks.</span>")
 		if(80 to 100)
 			to_chat(user, "<span class='notice'>It appears in perfect condition.</span>")
-	
+
+/obj/item/weapon/shield/riot/attackby(obj/item/I, mob/user, params)
+	. = ..()
```
Comment (Rohesie) [inline code/game/objects/items/weapons/shields.dm:48]: You should check if the attack was handled / cut short in a parent call.

**Pull request merged**